V2_MIGRATION_GUIDE.md

```diff
@@ -60,164 +63,129 @@ Additionally, any classes that implemented `ParameterizableRequest` or `AuthRequ
-
-#### DatabaseConnectionRequest
-- `public DatabaseConnectionRequest<T, U> addParameters(@NonNull Map<String, Object> parameters)`. Use `public DatabaseConnectionRequest<T, U> addParameters(@NonNull Map<String, String> parameters)` instead.
-- `public DatabaseConnectionRequest<T, U> addParameter(@NonNull String name, @NonNull String value)`. Use `public DatabaseConnectionRequest<T, U> addParameter(@NonNull String name, @NonNull String value)` instead.
-
-#### DelegationRequest
-- `public DelegationRequest<T> addParameters(@NonNull Map<String, Object> parameters)`. Use `public DelegationRequest<T> addParameters(@NonNull Map<String, String> parameters)` instead.
-- `public DelegationRequest<T> addParameter(@NonNull String name, @NonNull String value)`. Use `public DelegationRequest<T> addParameter(@NonNull String name, @NonNull String value)` instead.
-
-#### ProfileRequest
-- `public ProfileRequest addParameters(@NonNull Map<String, Object> parameters)`. Use `public ProfileRequest addParameters(@NonNull Map<String, String> parameters)` instead.
-- `public ProfileRequest addParameter(@NonNull String name, @NonNull String value)`. Use `public ProfileRequest addParameter(@NonNull String name, @NonNull String value)` instead.
-
-#### SignUpRequest
-- `public SignUpRequest addAuthenticationParameters(@NonNull Map<String, Object> parameters)`. Use `public SignUpRequest addAuthenticationParameters(@NonNull Map<String, String> parameters)` instead.
-- `public SignUpRequest addSignUpParameters(@NonNull Map<String, Object> parameters)`. Use `public SignUpRequest addSignUpParameters(@NonNull Map<String, String> parameters)` instead.
-- `public SignUpRequest addParameters(@NonNull Map<String, Object> parameters)`. Use `public SignUpRequest addParameters(@NonNull Map<String, String> parameters)` instead.
-- `public SignUpRequest addParameter(@NonNull String name, @NonNull String value)`. Use `public SignUpRequest addParameter(@NonNull String name, @NonNull String value)` instead.
-- `public SignUpRequest setDevice(@NonNull String device)`. Use `public Request<Credentials, AuthenticationException> addParameter("device", value)` instead.
-
-#### AuthenticationRequest
-- `public AuthenticationRequest addAuthenticationParameters(@NonNull Map<String, String> parameters)`. Use `public Request<Credentials, AuthenticationException> addParameters(@NonNull Map<String, String> parameters)` instead.
-- `public AuthenticationRequest setDevice(@NonNull String device)`. Use `public Request<Credentials, AuthenticationException> addParameter("device", value)` instead.
+- `addAuthenticationParameters(parameters)` has been removed. Use `addParameters(parameters)` instead.
+- `setDevice("{DEVICE}")`. Use `addParameter("{DEVICE}", "{VALUE}")` instead.
 
-### Constants removed
+#### `WebAuthProvider.Builder` removals and changes
```

Comment:
Seems like all below are removals. For one you call out "has been removed" but the rest don't have this context. I'd rename this title to just "removals" and remove the "has been removed" from #start.

If you rather keep the title this way, then you'd need to add "has been removed" to every line. But that sounds too repetitive. Take a look at the other titles below because I see the same pattern.

Reply:
Updated headings to only specify removals when applicable.